lib/generators/engine_cart/install_generator.rb

```diff
@@ -52,7 +52,12 @@ def add_gemfile_include
   # the below comes from engine_cart, a gem used to test this Rails engine gem in the context of a Rails app
   file = File.expand_path("Gemfile", ENV['ENGINE_CART_DESTINATION'] || ENV['RAILS_ROOT'] || File.expand_path("../spec/internal", __FILE__))
   if File.exist?(file)
-    eval_gemfile file
+    begin
+      eval_gemfile file
+    rescue Bundler::GemfileError => e
+       Bundler.ui.warn '[EngineCart] Skipping Rails application dependencies:'
+       Bundler.ui.warn e.message
+    end
```

Comment:
Are you sure you really get a Bundler::Gemfile error at this point in code in the problem cases we're seeing? I'd think bundler will evaluate the whole Gemfile first, and the error will be raised only when bundler tries to resolve dependencies. But I'd have to reproduce a problem case and see the stacktrace to be sure. The problem case is hard for me to repro because I've never fully understood what's going on, I just know it happens, and then I frustratedly delete things until it works again. @ndushay, can you repro a problem case to see if this really solves it?